### PR TITLE
Fixes for Edge

### PR DESCRIPTION
**Pull request by paulvarache**
- Create module files matching the Blockly file structure
 - Remove every circular dependency from the import.js file
 - Make each file import its depdencies instead of assuming they will be imported by import.js
 - Some ESLint things

### CODE REVIEW

**Comment by TeamKano** [comment]
<table>
    <tr>
        <th colspan="3">Coverage summary</th>
    </tr>
<tr>
        <td>Statements</td>
        <td>33.95%</td>
        <td>7516/22141</td>
    </tr>
<tr>
        <td>Branches</td>
        <td>10.14%</td>
        <td>1371/13520</td>
    </tr>
<tr>
        <td>Functions</td>
        <td>13.57%</td>
        <td>555/4089</td>
    </tr>
<tr>
        <td>Lines</td>
        <td>63.11%</td>
        <td>3036/4811</td>
    </tr>
</table>

*From Jenkins: KanoComponents/kwc-blockly/PR-52*

File: ext/blocks/functions.js [hunk 1]
```diff
@@ -156,12 +160,14 @@ Blockly.Blocks.function_call = {
...
     },
-    getFunctionCall () {
+    getFunctionCall() {
         return this.getFieldValue('NAME');
     },
-    updateShape () {
+    updateShape() {
         let connectionHistory = {},
-            input, funcDef, updateConnections;
+            input,
```
Comment (paulmck21) [inline ext/blocks/functions.js]: This isn't the usual pattern we use other places. Would it make more sense to declare funcDef on line 173 and input inside the foreach on line 182, as we don't reassign them? More in keeping with the style of the other codebases

File: ext/blocks/functions.js [hunk 2]
```diff
@@ -87,41 +92,40 @@ Blockly.Blocks.function_definition = {
         this._updateCallBlocks();
         this._updateArgBlocks();
     },
-    getToolbox () {
+    getToolbox() {
         let functionDef = this.workspace.functionsRegistry.getFunction(this.id),
```
Comment (paulmck21) [inline ext/blocks/functions.js]: const? Seems like you've done it elsewhere and maybe missed a few

File: ext/blocks/functions.js [hunk 3]
```diff
@@ -261,15 +267,15 @@ Blockly.Blocks.function_argument = {
             }
         });
     },
-    getFunctionDefinition () {
+    getFunctionDefinition() {
         return this.definitionWorkspace.functionsRegistry.getFunction(this.definitionBlock);
     },
-    updateShape () {
+    updateShape() {
         let funcDef = this.getFunctionDefinition(),
```
Comment (paulmck21) [inline ext/blocks/functions.js]: const?

File: ext/generators/javascript/functions.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
-Blockly.JavaScript.function_definition = (block) => {
-    let params    = [],
-        statement = Blockly.JavaScript.statementToCode(block, 'DO'),
-        name      = Blockly.JavaScript.variableDB_.getName(block.getFieldValue('NAME'), Blockly.Functions.NAME_TYPE),
-        returns   = Blockly.JavaScript.valueToCode(block, 'RETURNS');
+import { JavaScript } from '../../../javascript-module.js';
+
+JavaScript.function_definition = (block) => {
+    let params = [],
```
Comment (paulmck21) [inline ext/generators/javascript/functions.js]: Don't believe any of these are reassigned. const?

**Pull request merged**